tests/system_tests/external_interaction/callbacks/test_external_callbacks.py

```diff
@@ -0,0 +1,282 @@
+    assert np.all(zocalo_event["zocalo-centres_of_mass"][0] == [1, 2, 3])
+    assert np.all(zocalo_event["zocalo-bbox_sizes"][0] == [6, 6, 5])
+
+    # get dcids from zocalo device
+    dcid_reading = await zocalo_device.ispyb_dcid.read()
+    dcgid_reading = await zocalo_device.ispyb_dcgid.read()
+
+    dcid = dcid_reading["zocalo-ispyb_dcid"]["value"]
+    dcgid = dcgid_reading["zocalo-ispyb_dcgid"]["value"]
+
+    assert dcid != 0
+    assert dcgid != 0
+
+    # check the data in dev ispyb corresponding to this "collection"
+    ispyb_comment = fetch_comment(dcid)
+    assert ispyb_comment != ""
+    assert "Diffraction grid scan of 40 by 20 images" in ispyb_comment
+    assert "Zocalo processing took" in ispyb_comment
+    assert "Position (grid boxes) ['1', '2', '3']" in ispyb_comment
+    assert "Size (grid boxes) [6 6 5];" in ispyb_comment
+
+
+@pytest.mark.s03()
+def test_remote_callbacks_write_to_dev_ispyb_for_rotation(
+    RE_with_external_callbacks: RunEngine,
+    test_rotation_params: RotationInternalParameters,
+    fetch_comment,
+    fetch_datacollection_attribute,
+    undulator,
+    attenuator,
+    synchrotron,
+    s4_slit_gaps,
+    flux,
+    fake_create_devices,
+):
+    test_wl = 0.71
+    test_bs_x = 0.023
+    test_bs_y = 0.047
+    test_exp_time = 0.023
+    test_img_wid = 0.27
+
+    test_rotation_params.experiment_params.image_width = test_img_wid
+    test_rotation_params.hyperion_params.ispyb_params.beam_size_x = test_bs_x
+    test_rotation_params.hyperion_params.ispyb_params.beam_size_y = test_bs_y
+    test_rotation_params.hyperion_params.detector_params.exposure_time = test_exp_time
+    test_rotation_params.hyperion_params.ispyb_params.current_energy_ev = (
+        convert_angstrom_to_eV(test_wl)
+    )
+    test_rotation_params.hyperion_params.detector_params.current_energy_ev = (
+        convert_angstrom_to_eV(test_wl)
+    )
+
+    composite = RotationScanComposite(
+        attenuator=attenuator,
+        backlight=fake_create_devices["backlight"],
+        detector_motion=fake_create_devices["detector_motion"],
+        eiger=fake_create_devices["eiger"],
+        flux=flux,
+        smargon=fake_create_devices["smargon"],
+        undulator=undulator,
+        synchrotron=synchrotron,
+        s4_slit_gaps=s4_slit_gaps,
+        zebra=fake_create_devices["zebra"],
+    )
+
+    with patch("bluesky.preprocessors.__read_and_stash_a_motor", fake_read):
+        RE_with_external_callbacks(
+            rotation_scan(
+                composite,
+                test_rotation_params,
+            )
+        )
+
+    assert isfile("tmp/dev/hyperion_ispyb_callback.txt")
+    ispyb_log_tail = subprocess.run(
+        ["tail", "tmp/dev/hyperion_ispyb_callback.txt"],
+        timeout=1,
+        stdout=subprocess.PIPE,
+    ).stdout.decode("utf-8")
+
+    ids_re = re.compile(r"data_collection_ids=(\d+) data_collection_group_id=(\d+) ")
+    matches = ids_re.findall(ispyb_log_tail)
+
+    dcid = matches[0][0]
```

Comment:
Could: This seems brittle. Could we do something like some ispyb param as a UUID and do a search in the db for it?

Reply:
if it breaks I'll fix it that way :)

Reply:
Haha, fine